compiler/rustc_mir_build/src/build/matches/mod.rs

```diff
@@ -61,6 +63,10 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
                     break_scope,
                     variable_source_info,
                     declare_bindings,
+                    match cold_branch {
+                        Some(false) => Some(false),
+                        _ => None,
+                    },
```

Comment:
This deserves a comment so future readers don't have to figure out the recursive function behaviour again